docs/docs/guides/deployment/hydra-oauth2/hydra-oauth2.mdx

```diff
@@ -0,0 +1,62 @@
+---
+title: Use Ory Hydra as OAuth2 provider
+---
+
+A requirement to run your own Chainloop instance, is to have a way of authenticate users. One of the most common ways to do this is by using OAuth2.
+Ory Hydra is a popular OAuth2 provider that can be used to authenticate users in your Chainloop instance once connected to an external OIDC provider.
+
+## Pre-requisites
+
+To configure your Chainloop instance with Ory Hydra you'll need the following information:
+
+- Ory Hydra instance running
+- Access to your OICD provider configuration
+
+## Configure Ory Hydra
+
+First, you'll need to have an Ory Hydra instance running. You can follow the [official documentation](https://www.ory.sh/hydra/docs/) to set up your own instance.
+Then simply create a new OAuth2 client in your Ory Hydra instance. You can do this by running the following command:
+
+```bash
+$  hydra create oauth2-client --name "ACME Solutions"  --grant-type authorization_code,refresh_token --response-type code  --scope openid,email,profile  --redirect-uri http://CHAINLOOP_INSTANCE_URL/auth/callback --endpoint https://ORY_HYDRA_URL
+```
+
+Chainloop client will only request `openid`, `email` and `profile` scopes.
+
+Relevant information that can be noted from the command signature is:
+- `name`: The name of the OAuth2 client
+- `grant-type`: The grant type of the client it needs to be set `authorization_code` and `refresh_token`
```

Comment:
do we need refresh token? I do not think so.

Reply:
What do we have on dex?

Reply:
Not sure about dex but Auth0, which we use in prod seems to by default indeed add `refresh_token`. I guess it's fine although I do not think we keep any kind of refresh token in the client, we could potentially disable that grant and should be fine. 

For now, until we test that, I'd leave it as is

Thanks